src/core/a-entity.js

```diff
@@ -341,7 +387,7 @@ var proto = Object.create(ANode.prototype, {
         return;
       }
       // Component not yet initialized. Initialize component.
-      this.initComponent(name);
+      this.initComponent(name, false, newData);
```

Comment:
With https://github.com/aframevr/aframe/pull/1323/files#r59427013, you can just call `this.initComponent(name, newData)` without the mystery boolean.